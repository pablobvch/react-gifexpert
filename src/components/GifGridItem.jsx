import PropTypes from "prop-types"

export const GifGridItem = ({url, title}) => {
  return (
    <div className="card">
      <img src={url} alt={title}></img>
      <p>{title}</p>
    </div>
  )
}

GifGridItem.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}