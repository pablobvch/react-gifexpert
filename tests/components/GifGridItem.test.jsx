import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';

describe('GifGridItem tests', () => { 
        const title = "Title test"
        const url = "http://www.google.com/"
        
    test('match with snapshot', () => {
        
        const { container } = render(<GifGridItem title={ title } url = {url} />);
        expect( container ).toMatchSnapshot();
    });

    test('must show the image with the url', () => {
        render(<GifGridItem title={ title } url = {url} />);
        //expect(screen.getByRole("img").src).toBe(url)
        const {src,alt} = screen.getByRole("img")
        expect(src).toBe(url)
        expect(alt).toBe(title)
    });
    test('must show the title in the component', () => {
        render(<GifGridItem title={ title } url = {url} />);
        expect(screen.getByText(title)).toBeTruthy()
    });
 })