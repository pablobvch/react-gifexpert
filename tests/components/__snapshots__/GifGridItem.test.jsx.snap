// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title test"
      src="http://www.google.com/"
    />
    <p>
      Title test
    </p>
  </div>
</div>
`;
